src/Tinfoil/KDF/Scrypt.hs

```diff
@@ -27,10 +27,10 @@ defaultParams = scryptParams 16 8 12 -- N = 2^16, r = 8, p = 12
 salt :: IO Entropy
 salt = entropy 32
 
-verifyNoCredential :: ScryptParams -> IO (Maybe Verified)
-verifyNoCredential p = do
+verifyNoCredential :: ScryptParams -> Credential -> IO (Maybe Verified)
+verifyNoCredential p c = do
   e <- salt
```

Comment:
is there a sensible way to get rid of this salt generation? I feel like verify takes long enough that we shouldn't be doing things we don't absolutely need to (i.e. isn't adding to strength of the hash). I think that using the user provided input is the right thing, but a hard coded salt of the right length doesn't affect that, we could then drop the salt from both call paths then.